DigiMeSDK/Core/Classes/Security/DMEJWTManager.swift

```diff
@@ -0,0 +1,444 @@
+        guard
+            let publicKeyBase64 = retrieveAuthorityPublicKeyIfValid(),
+            let publicKey = convertKeyString(publicKeyBase64) else {
+            print("Error creating RSA public key")
+            return nil
+        }
+        
+        // validation
+        let verifier = JWTVerifier.ps512(publicKey: publicKey)
+        let isVerified = JWT<PayloadValidateAuthJWT>.verify(authToken, using: verifier)
+        
+        guard isVerified else {
+            return nil
+        }
+        
+        let decoder = JWTDecoder(jwtVerifier: verifier)
+        let jwt = try? decoder.decode(JWT<PayloadValidateAuthJWT>.self, fromString: authToken)
+        guard
+            let accessToken = jwt?.claims.accessToken,
+            let refreshToken = jwt?.claims.refreshToken,
+            let expiresTimestamp = jwt?.claims.expiresTimestamp else {
+            return nil
+        }
+        
+        let oauthObj = DMEOAuthObject()
+        oauthObj.accessToken = accessToken
+        oauthObj.refreshToken = refreshToken
+        oauthObj.expiresOn = Date(timeIntervalSince1970: expiresTimestamp)
+        
+        return oauthObj
+    }
+    
+    @objc public func createDataTriggerToken(_ accessToken: String, appId: String, contractId: String, sessionKey: String, privateKeyBase64: String, publicKeyBase64: String?) -> String? {
+        let claims = PayloadDataTriggerJWT()
+        claims.accessToken = accessToken
+        claims.clientId = "\(appId)_\(contractId)"
+        claims.nonce = (DMECryptoUtilities.getRandomBytes(withLength: 16) as NSData).hexString()
+        // NB! this redirect schema must exist in the CA contract definition (Cadmium/Neon tool), otherwise this request will fail!
+        claims.redirectUrl = "digime-ca-\(appId)"
+        claims.sessionKey = sessionKey
+        claims.timestamp = NSDate().timeIntervalSince1970 * 1000.0
+        
+        guard let privateKey = convertKeyString(privateKeyBase64) else {
+            print("Error creating RSA key")
+            return nil
+        }
+
+        // signing
+        var jwt = JWT(header: header, claims: claims)
+        let signer = JWTSigner.ps512(privateKey: privateKey)
+        guard let signedToken = try? jwt.sign(using: signer) else {
+            print("Error signing our test token")
+            return nil
+        }
+
+        // validation
+        guard
+            let publicKeyBase64 = publicKeyBase64,
+            let publicKey = convertKeyString(publicKeyBase64) else {
+                return signedToken
+        }
+        
+        let verifier = JWTVerifier.ps512(publicKey: publicKey)
+        let isVerified = JWT<PayloadDataTriggerJWT>.verify(signedToken, using: verifier)
+
+        return isVerified ? signedToken : nil
+    }
+    
+    @objc public func createRefreshToken(_ refreshToken: String, appId: String, contractId: String, privateKeyBase64: String, publicKeyBase64: String?) -> String? {
```

Comment:
Please add documentation comment:

```swift
    /// Creates refresh JWT which can be used to refresh an associated access token
    /// - Parameters:
    ///   - refreshToken: <#refreshToken description#>
    ///   - appId: <#appId description#>
    ///   - contractId: <#contractId description#>
    ///   - privateKey: private key in base 64 format
    ///   - publicKey: public key in base 64 format
```

Please also update the signature to:
`refreshJwt(from refreshToken: String, appId: String, contractId: String, privateKey: String, publicKey: String?) -> String?`

**Also**, please add a `guard` statement to verify the `privateKey` is in correct format

**Lastly** please make this a `class` function